viewer/packages/camera-manager/src/ComboControls.ts

```diff
@@ -583,37 +591,58 @@ export default class ComboControls extends EventDispatcher {
     const cameraDirection = reusableVector3;
     let radius = distToTarget + deltaDistance;
 
-    if (radius < minDistance) {
-      radius = minDistance;
-      if (dynamicTarget) {
-        // push targetEnd forward
-        reusableCamera.getWorldDirection(cameraDirection);
-        targetEnd.add(cameraDirection.normalize().multiplyScalar(Math.abs(deltaDistance)));
-      } else {
-        // stops camera from moving forward
-        deltaDistance = distToTarget - radius;
+    if (moveTarget) {
+      // move target together with the camera (for 'w' and 's' keys movement)
+      reusableCamera.getWorldDirection(cameraDirection);
+      targetEnd.add(cameraDirection.normalize().multiplyScalar(-deltaDistance));
+    } else {
+      // behaviour for scrolling with mouse wheel
+      if (radius < minDistance) {
+        radius = minDistance;
+        if (dynamicTarget) {
+          // push targetEnd forward
+          reusableCamera.getWorldDirection(cameraDirection);
+          targetEnd.add(cameraDirection.normalize().multiplyScalar(Math.abs(deltaDistance)));
+        } else {
+          // stops camera from moving forward only if target became close to scroll target
+          if (this.scrollTarget.clone().sub(this.target).length() < 0.2) deltaDistance = distToTarget - radius;
+        }
       }
```

Comment:
Try to clean this up if you can to avoid so many indent-blocks. Either by precomputing state, or extracting into private functions (or both).